### ISSUE TITLE
Massive memory leak

### ISSUE THREAD

**Issue by GrahamCampbell** [madewithlove/illuminate-psr-cache-bridge#5]
You cache forever rather than for a specific amount of time. This will fill the cache.

**Comment by danhunsaker** [comment]
I don't think that counts as a memory leak, nor an issue with this package.  To cache something forever, the developer has to specify forever as the duration.  If the developer doesn't expect their cache to fill up as a result, or at least to be very large, that developer needs to relearn how caching works, as well as the definition of forever.  Since each request to add items to the cache allows the developer to specify a duration, it's up to the developer themselves to specify something sane, based on what they're caching at the time.

And as for it being a memory leak, even ignoring that not all Laravel cache drivers even use RAM for storage, it's not a leak unless things stay reserved even after they're no longer actually in use - as long as that cached value is still stored and accessible there, that cache location is still in use.  Therefore, not a leak.

**Comment by GrahamCampbell** [comment]
In redis, it really does mean forever, not just a long time. This is a memory leak, and will result in many GB of trash. Why not set the expiry to 30 days or something, or allow user configuration via a constructor param?

**Comment by danhunsaker** [comment]
I use Redis for my cache as well, so I'm aware of its expiration setup.  Thing is, if you tell the library to cache something without telling it when that something should expire, _you_ are responsible for the fact that it never will expire.  This lib is following both PSR-6 and Laravel requirements and conventions.  It's up to developers to learn what those are and how to use them properly.

TL;DR - Set expirations on your CacheItem objects, because PSR-6 and Laravel both say forever is what you should get otherwise.

**Comment by GrahamCampbell** [comment]
Your library always calls the forever method though?

**Comment by GrahamCampbell** [comment]
> TL;DR - Set expirations on your CacheItem objects

If we do set an expiration, it's still cached forever? It just discards it on read later after it "expires", but if we never read it again, it's stuck there forever isn't it?

**Comment by GrahamCampbell** [comment]
Maybe I'm missing something?

**Comment by danhunsaker** [comment]
For one, not my lib; I'm just a user, like you.

For two, no, it doesn't.  It only calls forever if you don't set an expiration on the CacheItem.  Feel free to review the code and point out where I'm mistaken in this.

**Comment by GrahamCampbell** [comment]
Oh, the very last occurrence of `$this->repository->` in the file calls put! I completely missed that. I was just doing a quick security review of this before using it in production and thought it didn't expire ever. Sorry for the false alarm!
